src/researchhub_case/views/author_claim_case_view.py

```diff
@@ -127,13 +132,27 @@ def _post_author_claim_cases_for_mods(self, request):
         try:
             request_data = request.data
             update_status = request_data['update_status']
-            if (update_status not in ['APPROVED', 'DENIED']):
-                return Response('Base update status', status=400)
-
             case_id = request_data['case_id']
             case = AuthorClaimCase.objects.get(id=case_id, status=OPEN)
-            case.status = APPROVED if update_status == "APPROVED" else DENIED
-            case.save()
-            return Response('Success', status=200)
+
+            if update_status == APPROVED:
+                if case.target_paper is None:
+                    return Response('Cannot approve. No paper id found.', status=500)
+                else:
+                    move_paper_to_author(case.target_paper, case.requestor.author_profile)
+                    case.status = update_status
+                    case.save()
+                    print('update_status', update_status)
+                    after_approval_flow.apply_async((case_id,), priority=2)
+                    return Response('Success', status=200)
+            elif update_status == DENIED:
+                notify_user = request_data['notify_user']
+                after_rejection_flow.apply_async((case_id, notify_user), priority=2)
+                case.status = update_status
+                case.save()
+                return Response('Success', status=200)
+            else:
+                return Response('Unrecognized status', status=400)
         except (KeyError, TypeError) as e:
+            print('e', e)
             return Response(e, status=400)
```

Comment:
This comment is unrelated to this line, but for this viewset, we should add the permission class `CreateOrReadOnly` from `utils/permissions.py` because bad faith actors can use this endpoint to modify existing claim cases.